v0/destinations/vero/transform.js

```diff
@@ -0,0 +1,124 @@
+const set = require("set-value");
+const get = require("get-value");
+const {
+  defaultRequestConfig,
+  CustomError,
+  constructPayload,
+  getValueFromMessage,
+  isDefinedAndNotNull,
+  removeUndefinedAndNullAndEmptyValues
+} = require("../../util");
+const { EventType } = require("../../../constants");
+const { CONFIG_CATEGORIES, MAPPING_CONFIG } = require("./config");
+
+// This function handles the common response payload for the supported calls
+const commonResponseBuilder = (message, category, destination) => {
+  const payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+  const authToken = get(destination, "Config.authToken");
+  set(payload, "auth_token", authToken);
+
+  // channels is defined only for Identify call
+  if (category.type === EventType.IDENTIFY) {
+    const channels = get(payload, "channels");
+    const address = get(channels, "address");
+    const platform = get(channels, "platform");
+
+    // Vero only accepts a valid channels object - address, platform and type should be defined and not null
+    if (isDefinedAndNotNull(address) && isDefinedAndNotNull(platform)) {
+      set(payload, "channels.type", category.channelType);
+    }
+    // Added this block because channels & channels.device can be defined and not null, which should be removed
+    else if (isDefinedAndNotNull(channels)) {
+      delete payload.channels;
+    }
```

Comment:
Its mentioned channels is optional field. Also when `address` and `platform` is not present why cant we ommit channel

Reply:
Test out if the channel object is forming correctly or not

Reply:
Tested and updated